### PR TITLE
feat(volume) UI add volume encryption support

### PR DESCRIPTION
**Pull request by smallteeths**
image: wangsiye/longhorn-ui:e68aeb2

https://github.com/longhorn/longhorn/issues/2841

Signed-off-by: Siye Wang <siye@rancher.com>

### CODE REVIEW

File: src/routes/backup/RestoreBackup.js [hunk 1]
```diff
@@ -116,6 +116,12 @@ const modal = ({
             { backingImages.map(backingImage => <Option key={backingImage.name} value={backingImage.name}>{backingImage.name}</Option>) }
           </Select>)}
         </FormItem>
+        <FormItem label="Encrypted" {...formItemLayout}>
+          {getFieldDecorator('encrypted', {
+            valuePropName: 'encrypted',
+            initialValue: false,
```
Comment (innobead) [inline src/routes/backup/RestoreBackup.js:122]: Just a check. It seems users need to specify the secret info again when restoring a backup, right? (maybe there are some place to improve this part in the future)

**Review by innobead — APPROVED** [review]
LGTM. Just one question about restoring a backup.

I leave to @joshimoo check again before merging.

File: src/routes/backup/RestoreBackup.js (hunk 1 above)
Comment (smallteeths) [reply src/routes/backup/RestoreBackup.js:122]: Yes, if we need to get the corresponding volume.encrypted this needs to be considered a bit more. If the volume has been deleted we won't get the encrypted property. I can consider setting the initial value for it if needed.

**Review by joshimoo — APPROVED** [review]
LGTM at the moment we don't store the encrypted flag as part of a backup.
So a restore requires manually specifying that this is an encrypted volume.
We also need to manually specify the secret for the PV ref when manually creating a PV for a restored volume. 

In the future we can include that encrypted field or store the kubernetes PV as part of a volume backup. To help automate some of these things.

Alternatively allow specifying a storage class for a restored volume, that way we would get all necessary information for the volume context from the storage class. 

One important thing that we cannot store is the actual secret content, since then we would be exposing the encryption keys for a volume.

**Pull request merged**